anndata/_io/__init__.py

```diff
@@ -9,5 +9,22 @@
     read_zarr,
     read_h5ad,
 )
-from .write import write_csvs, write_loom, _write_h5ad, write_zarr
+from .write import write_csvs, write_loom, write_zarr
+from .write import _write_h5ad  # noqa: F401
 from . import h5ad
+
+__all__ = [
+    "read_csv",
+    "read_excel",
+    "read_umi_tools",
+    "read_hdf",
+    "read_loom",
+    "read_mtx",
+    "read_text",
+    "read_zarr",
+    "read_h5ad",
+    "write_csvs",
+    "write_loom",
+    "write_zarr",
+    "h5ad",
+]
```

Comment:
So, two things here:

* I don't think we should define namespaces for private modules
* I am up for a public `io` module, but I think we should do that in a separate PR

```suggestion
```